test/client/fixtures/sandbox/node/window-test.js

```diff
@@ -16,6 +17,144 @@ test('window.onerror setter/getter', function () {
+                        return false;
+                    });
+
+                    return testPreventing();
+                })
+                .then(function () {
+                    setProperty(window, 'onunhandledrejection', function (event) {
+                        event.preventDefault();
+                    });
+
+                    return testPreventing();
+                })
+                .then(function () {
+                    setProperty(window, 'onunhandledrejection', null);
+
+                    window.addEventListener('unhandledrejection', function onUnhandledRejection (event) {
+                        event.preventDefault();
+                        window.removeEventListener('unhandledrejection', onUnhandledRejection);
+                    });
+
+                    return testPreventing();
+                });
+        });
+
+        test('should convert an unhandled rejection reason to string', function () {
+            var testMsg = function (err) {
+                return new hammerhead.Promise(function (resolve) {
+                    hammerhead.on(hammerhead.EVENTS.unhandledRejection, function onUnhandledRejection (event) {
+                        hammerhead.off(hammerhead.EVENTS.unhandledRejection, onUnhandledRejection);
+                        resolve(event.msg);
+                    });
+
+                    /*eslint-disable no-new*/
+                    new Promise(function () {
+                        throw err;
+                    });
+                    /*eslint-enable no-new*/
+                });
+            };
+
+            return testMsg(1)
+                .then(function (msg) {
+                    strictEqual(msg, '1');
+                    return testMsg(new Error('error message'));
+                })
+                .then(function (msg) {
+                    strictEqual(msg, 'error message');
+                    return testMsg(new TypeError('type error'));
+                })
+                .then(function (msg) {
+                    strictEqual(msg, 'type error');
+                    return testMsg('string message');
+                })
+                .then(function (msg) {
+                    strictEqual(msg, 'string message');
+                    return testMsg(true);
+                })
+                .then(function (msg) {
+                    strictEqual(msg, 'true');
+                    return testMsg({ a: 1 });
+                })
+                .then(function (msg) {
+                    strictEqual(msg, '[object Object]');
+                    return testMsg(void 0);
+                })
+                .then(function (msg) {
+                    strictEqual(msg, 'undefined');
+                    return testMsg(function () {
+                    });
+                })
+                .then(function (msg) {
+                    strictEqual(msg, '[object Function]');
+                });
```

Comment:
Add test for `Symbol` type.
Reorder check as : null, undefined, string, number, boolean, symbol, object, function